backend/src/resolvers/comments.js

```diff
@@ -2,44 +2,47 @@ import { neo4jgraphql } from 'neo4j-graphql-js'
 import { UserInputError } from 'apollo-server'
 
 const COMMENT_MIN_LENGTH = 1
-export default {
-  Query: {
-    CommentByPost: async (object, params, context, resolveInfo) => {
-      const { postId } = params
-
-      const session = context.driver.session()
-      const transactionRes = await session.run(`
-          MATCH (comment:Comment)-[:COMMENTS]->(post:Post {id: $postId})
-          RETURN comment {.id, .contentExcerpt, .createdAt} ORDER BY comment.createdAt ASC`, {
-        postId
-      })
+const NO_POST_ERR_MESSAGE = 'Comment cannot be created without a post!'
 
-      session.close()
-      let comments = []
-      transactionRes.records.map(record => {
-        comments.push(record.get('comment'))
-      })
-
-      return comments
-    }
-  },
+export default {
   Mutation: {
     CreateComment: async (object, params, context, resolveInfo) => {
       const content = params.content.replace(/<(?:.|\n)*?>/gm, '').trim()
+      const { postId } = params
+      // Adding relationship from comment to post by passing in the postId,
+      // but we do not want to create the comment with postId as an attribute
+      // because we use relationships for this. So, we are deleting it from params
+      // before comment creation.
+      delete params.postId
 
       if (!params.content || content.length < COMMENT_MIN_LENGTH) {
         throw new UserInputError(`Comment must be at least ${COMMENT_MIN_LENGTH} character long!`)
       }
-      const { postId } = params
-      delete params.postId
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-04T09:05:12Z" title="Saturday, May 4th 2019, 11:05:12 am +02:00">May 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -31,15 +11,15 @@ export default {
         throw new UserInputError(`Comment must be at least ${COMMENT_MIN_LENGTH} character long!`)
       }
       const { postId } = params
-      delete params.postId
```


Because you don’t delete the `postId` from the `params` you store it now in the comment.
Is this what you want and if so, why? @mattwr18

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-04T12:16:05Z" title="Saturday, May 4th 2019, 2:16:05 pm +02:00">May 4, 2019</time>_

---


yes, well let me tell you about that, hehe
I can't remember exactly when on the journey, but I thought that I needed to add it to be able to use the `Comment` query but search by the `postId`, so I deleted that line, however I figured out a different implementation that is better, in my opinion, then forgot to add it back. 
good catch!